### PR TITLE
[Hub Generated] Review request for Microsoft.Billing to add version preview/2018-11-01-preview

### PR DESCRIPTION
**Pull request by msft-adrianma**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/msft-adrianma/azure-rest-api-specs/patchBillingAccount-billing-Microsoft.Billing-2018-11-01-preview...master/:tab?).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by wilcobmsft — APPROVED** [review]

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 1]
```diff
@@ -106,6 +106,63 @@
...
+          },
+          {
+            "$ref": "#/parameters/billingAccountNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BillingAccount"
```
Comment (RyanBensonMSFT) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: Patch typically doesn't require the same input as Put. You should create a new type definition which contains only the 'patchable' properties of a BillingAccount, the convention we typically use would name this new definition 'BillingAccountUpdateParameters'. 

Take a look at the follow swagger for an example, search for StorageAccountUpdateParameters.
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/Microsoft.Storage/stable/2018-11-01/storage.json

For more info on how Patch should work:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#patch-resource

Comment (msft-adrianma) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: updated to use updateparameters. Currently we only allow update of billing account address.

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6398 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6398/commits/d3a58bec5c1fe11638729af5df36c2add8e2f3e3

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/UpdateBillingAccount.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        "postalCode": "98052",
+        "region": "WA",
+        "country": "US"
+      }
+    }
+  },
+  "responses": {
+    "202": {
+      "headers": {
+        "Location": "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/patchOperations/65e1f2bf-c31c-4b89-a599-25d9d4172af9?api-version=2018-11-01-preview",
```
Comment (RyanBensonMSFT) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/UpdateBillingAccount.json]: You shouldn't include location with 202. For Patch and re-Put, to get the status of the change they just Get the resource and provisioningState should be set to the status. If you want a richer status API, use Azure-AsyncOperation.

https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creating-or-updating-resources

Comment (msft-adrianma) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/UpdateBillingAccount.json]: updated to use Azure-AsyncOperation

Comment (msft-adrianma) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/UpdateBillingAccount.json]: updated.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 3]
```diff
@@ -106,6 +106,63 @@
...
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/BillingAccount"
+            }
+          },
+          "202": {
+            "description": "Accepted. Billing account update is in progress.",
+            "headers": {
+              "Location": {
```
[the remark this answers is not in the document]
Comment (msft-adrianma) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: updated.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/UpdateBillingAccount.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        "postalCode": "98052",
+        "region": "WA",
+        "country": "US"
+      }
+    }
+  },
+  "responses": {
+    "202": {
+      "headers": {
+        "Azure-AsyncOperation": "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/patchOperations/65e1f2bf-c31c-4b89-a599-25d9d4172af9?api-version=2018-11-01-preview",
```
[the remark this answers is not in the document]
Comment (msft-adrianma) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/examples/UpdateBillingAccount.json]: updated

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 5]
```diff
@@ -3743,6 +3793,15 @@
         }
       }
     },
+    "BillingAccountUpdateProperties": {
+      "description": "The properties of the billing account that can be updated.",
+      "properties": {
+        "address": {
```
Comment (RyanBensonMSFT) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: One final thing:
The body of patch should replicate the resource but only include the fields being patched. In this case you have:
{
  'address':...
}
but it should be:
{
  'properties':{
    'address': ...
  }
}

**Comment by shahabhijeet** [comment]
This is good to be merged as soon as CI is green

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Pull request merged**